influxql/iterator.go

```diff
@@ -431,6 +437,22 @@ func newIteratorOptionsStmt(stmt *SelectStatement) (opt IteratorOptions, err err
 	return opt, nil
 }
 
+// MergeSorted returns true if the options require a sorted merge.
+// This is only needed when the expression is a variable reference or there is no expr.
+func (opt IteratorOptions) MergeSorted() bool {
```

Comment:
While it's a couple of extra lines, I think it's better to check for nil and return true before attempting a cast. It makes it more clear what's going on and doesn't require a reflection check when the expression doesn't exist.

I don't know if that type cast though actually have any negative effect on the performance.

Something like this works for me:

```
if opt.Expr == nil {
    return true
}
_, ok := opt.Expr.(*VarRef)
return ok
```